service/handlers.go

```diff
@@ -166,23 +159,28 @@ func GetUserData(c *gin.Context) {
 
 	r := data_store.NewUserDater()
 	if r == nil {
-		c.JSON(http.StatusInternalServerError, INIT_USERDATA_FAIL)
+		c.JSON(http.StatusInternalServerError, outputErrorFormat(INIT_USERDATA_FAIL))
 	}
 
-	response := r.GetUserActivity(username)
+	response, err := r.GetUserActivity(username)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, outputErrorFormat(err.Error()))
+		return
+	}
 	if response == nil {
+		c.JSON(http.StatusInternalServerError, outputErrorFormat("Got Nil response"))
 		return
 	}
 
 	userPath1 := fmt.Sprintf("%v", response[0])
 	userPath2 := fmt.Sprintf("%v", response[1])
 	userPath3 := fmt.Sprintf("%v", response[2])
 
-	userData := map[string]string{"path1": userPath1}
+	userData := map[string]string{"last action": userPath1}
 	if userPath2 != "" {
-		userData["path2"] = userPath2
+		userData["2nd last action"] = userPath2
 		if userPath3 != "" {
```

Comment:
you don't need this condition to be nested...

Reply:
userPath3 wont exist if 2 doesn't. so it can be nested and save some checks